collect_app/src/main/java/org/odk/collect/android/preferences/screens/ProjectPreferencesFragment.kt

```diff
@@ -68,14 +69,26 @@ class ProjectPreferencesFragment :
         findPreference<Preference>(CHANGE_ADMIN_PASSWORD_PREFERENCE_KEY)!!.onPreferenceClickListener = this
         findPreference<Preference>(PROJECT_MANAGEMENT_PREFERENCE_KEY)!!.onPreferenceClickListener = this
         findPreference<Preference>(ACCESS_CONTROL_PREFERENCE_KEY)!!.onPreferenceClickListener = this
+
+        if (inFormEntry) {
+            findPreference<Preference>(PROJECT_MANAGEMENT_PREFERENCE_KEY)!!.also {
+                it.isEnabled = false
+                it.setSummary(org.odk.collect.strings.R.string.setting_not_available_during_form_entry)
+            }
+        }
     }
 
     override fun onPreferenceClick(preference: Preference): Boolean {
         if (MultiClickGuard.allowClick(javaClass.name)) {
             when (preference.key) {
                 PROTOCOL_PREFERENCE_KEY -> displayPreferences(ServerPreferencesFragment())
                 PROJECT_DISPLAY_PREFERENCE_KEY -> displayPreferences(ProjectDisplayPreferencesFragment())
-                USER_INTERFACE_PREFERENCE_KEY -> displayPreferences(UserInterfacePreferencesFragment())
+                USER_INTERFACE_PREFERENCE_KEY -> {
+                    val fragment = UserInterfacePreferencesFragment()
+                    fragment.arguments =
```

Comment:
Why here do you use args and in the case above you pass the value via constructor using fragmentfactory?

Reply:
I've used the constructor in the case where the Fragment is declared in the XML layout and an arg in the case where the Fragment is created programmatically. I didn't want to rework the Fragment in the latter case to use a `FragmentFactory` as it felt like beyond the scope of fixing this bug.